### PR TITLE
fix: randomPort option as false by default

### PR DESCRIPTION
**Pull request by sajixavier**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ exports.default = class GeckoService {
         this.outputDir = options.logs || config.geckoDriverLogs || config.outputDir
         this.logFileName = options.logFileName || DEFAULT_LOG_FILENAME
         this.args = options.args || config.geckoDriverArgs || []
-        this.randomPort = options.useRandomPort || config.geckoDriverRandomPort || true
+        this.randomPort = options.useRandomPort || config.geckoDriverRandomPort || false
```
Comment (christian-bromann) [inline src/index.js:55]: I guess we have to check whether the port is set
```suggestion
        this.randomPort = options.useRandomPort || config.geckoDriverRandomPort || Boolean(options.port)
```

**Comment by christian-bromann** [comment]
@sajixavier can you check why the test fail?

**Comment by sajixavier** [comment]
it is because of 'maxInstances' value in wdio.conf.ts. If that was set it as 1 then this will work. (same port used by instances) So to make it work, we need to keep the value as true by default. With this change whenever there is a custom port number this.randomPort will be false.

**Comment by christian-bromann** [comment]
What would you suggest then? I guess this was the reason for having it set to true by default. Geckodriver can only handle one session so once you use this service with two session it will fail with the current patch.

One suggestion is to allow users to set the service option through capabilities, e.g. `wdio:geckodriverService` and start the instance in `beforeSession`.

**Comment by sajixavier** [comment]
Can we use maxInstances value to decide whether to use randomPort (use randomPort if maxIntances is more than 1). I am not really familiar with solution that you have mentioned.

**Comment by christian-bromann** [comment]
> I am not really familiar with solution that you have mentioned.

Right now the port is defined as service option, e.g.
```ts
// wdio.conf.js
export.config = {
    services: [
        [
            'geckodriver',
            // service options
            {
                port: 4444
            }
        ]
    ],
};
```
As we know this choice is flawed given that Geckodriver can only run one instance and if you test with multiple sessions you can't assign the same port for multiple Geckodriver instances. That is why I would suggest to move the port definition to the capabilities, e.g.:

```ts
capabilities: [{
    browserName: 'firefox',
    'wdio:geckodriverService': { port: 4444 }
}]
```

The reason why it should be `wdio:geckodriverService` is because capability names are defined by the WebDriver protocol. It allows custom capabilities in the format of `<vendor>:<optionName>`. This means we have to move the logic from the service launcher into the service and start geckodriver in the `beforeSession` hook. We should then read the `wdio:geckodriverService` option from the capabilities and start either on provided port or on a random one.

Let me know if this is not clear enough,

**Comment by christian-bromann** [comment]
I just pushed a commit that will change the current behavior and will always use a random port if the port option is omitted.

**Pull request closed without merging**